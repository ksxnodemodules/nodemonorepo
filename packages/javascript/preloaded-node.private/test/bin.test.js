'use strict'
const path = require('path')
const {spawnSync} = require('child_process')
const xjest = require('extra-jest')
const {bin} = require('../index')
const script = path.resolve(__dirname, 'data/main.js')

const {
  status,
  signal,
  error,
  stdout,
  stderr
} = spawnSync(
  'node',
  [bin, script],
  {
    encoding: 'utf8'
  }
)

const testfunc = xjest.snap.unsafe({
  status,
  signal,
  error: error ? String(error) : null,
  stdout: stdout || '(Empty)',
  stderr: stderr || '(Empty)'
})

it('matches snapshot', testfunc)
