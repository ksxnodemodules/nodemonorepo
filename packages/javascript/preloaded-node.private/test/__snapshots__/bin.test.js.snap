// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
"
error: null
signal: null
status: 123
stderr: (Empty)
stdout: |
  { TypeScript: 'Hello World !!' }

"
`;
